docs/connectors/sinks/delta.md

```diff
@@ -0,0 +1,125 @@
+small updates, which is normal and even preferable for output transports like
+Kafka; however it can cause problems for the Delta Lake format by creating a large
+number of small files.
+
+The output buffer mechanism is designed to solve this problem by decoupling the
+rate at which the pipeline pushes changes to the output transport from the rate
+of input changes.  It works by accumulating updates inside the pipeline
+for up to a user-defined period of time or until accumulating a user-defined number
+of updates and writing them to the Delta Table as a small number of large files.
+
+The following parameters are used to configure the output buffer
+
+* `enable_output_buffer` - Enable output buffer.
+
+* `max_output_buffer_time_millis` - Maximum time in milliseconds data is kept
+   in the output buffer.
+
+   When not specified, data is kept in the buffer indefinitely until one of
+   the other trigger conditions is satisfied.  When this option is
+   set the buffer will be flushed at most every
+   `max_output_buffer_time_millis` milliseconds.
+
+   NOTE: this configuration option requires the `enable_output_buffer` flag
+   to be set.
+
+* `max_output_buffer_size_records` - Maximum number of updates to be kept in
+   the output buffer.
+
+   This parameter bounds the maximal size of the buffer.  
+   Note that the size of the buffer is not always equal to the
+   total number of updates output by the pipeline. Updates to the
+   same record can overwrite or cancel previous updates.
+
+   When not specified, the buffer can grow indefinitely until one of
+   the other trigger conditions is satisfied.
+
+   NOTE: this configuration option requires the `enable_output_buffer` flag
+   to be set.
+
+:::note
+
+when the `enable_output_buffer` flag is set, at least one of
+`max_output_buffer_time_millis` or `max_output_buffer_size_records` must be
+specified.
+
+:::
+
+## Limitations
+
+This connector currently only appends new records to the Delta Table.  Deletions
+output by the pipeline are discarded.  The reason for this limitation is that the
+Delta Lake format only supports appending new data to a table in streaming mode.
+
+## Example usage
+
+### curl
+
+Create a Delta Lake output connector that writes a stream of updates to a table
+stored in an S3 bucket, truncating any existing contents of the table.
+
+```bash
+curl -i -X PUT http://localhost:8080/v0/connectors/delta-output-test \
+-H "Authorization: Bearer <API-KEY>" \
+-H 'Content-Type: application/json' \
+-d '{
+  "description": "Delta table output connector example",
+  "config": {
+    "transport": {
+      "name": "delta_table_output",
+      "config": {
+        "uri": "s3://feldera-fraud-detection-demo/feature_train",
+        "mode": "truncate",
+        "aws_access_key_id": <AWS_ACCESS_KEY_ID>,
+        "aws_secret_access_key": <AWS_SECRET_ACCESS_KEY>,
+        "aws_region": "us-east-1"
+      },
+      "enable_output_buffer": true,
```

Comment:
Is the buffer setting common to all output connectors? If so, we need a separate documentation section for it.

Reply:
Yes, and yes, but it doesn't make a whole lot of sense for the other connectors we have now. I'll add the section.